low-level/files/opt/seagate/sspl/setup/sspl_post_install.py

```diff
@@ -0,0 +1,156 @@
+
+import os
+import sys
+import errno
+import traceback
+import subprocess
+import shutil
+import distutils.dir_util
+
+# Add the top level directories
+sys.path.insert(0, '/opt/seagate/cortx/sspl/low-level')
+from framework.base.sspl_constants import (PRODUCT_FAMILY, REPLACEMENT_NODE_ENV_VAR_FILE, PRODUCT_NAME,
+    SSPL_BASE_DIR, file_store_config_path, PRODUCT_BASE_DIR)
+from files.opt.seagate.sspl.setup.sspl_setup import Cmd as SSPLSetup
+
+class SSPLPostInstallCmd:
+    "SSPL Post Install"
+
+    name = "sspl_post_install"
+
+    def __init__(self, args: list):
+        self.args = args
+        self._script_dir = os.path.dirname(os.path.abspath(__file__))
+        self.RSYSLOG_CONF="/etc/rsyslog.d/0-iemfwd.conf"
+        self.RSYSLOG_SSPL_CONF="/etc/rsyslog.d/1-ssplfwd.conf"
+        self.PACEMAKER_INSTALLATION_PATH="/lib/ocf/resource.d/seagate/"
+
+    def usage(self):
+        sys.stderr.write(
+            f"{self.name} [[-p <LDR_R1|LDR_R2>] [-e <DEV|PROD>] [-c|--rmq-cluster <true|false>]]\n"
+            "-p Product to be configured\n"
+            "-e Environment\n"
+            "-c Need rmq cluster? (true or false)\n"
+            )
+        sys.exit(1)
+    
+    def process(self):
+        if not self.args or len({'-p', '-e', '-c'} & set(self.args)) == 0:
+            self.usage()
+
+        PRODUCT = ""
+        ENVIRONMENT = ""
+        RMQ_CLUSTER = "true"
+        i = 0
+        while i < len(self.args):
+            try:
+                if self.args[i] == '-p':
+                    PRODUCT = self.args[i+1]
+
+                elif self.args[i] == '-e':
+                    ENVIRONMENT = self.args[i+1]
+
+                elif self.args[i] == '-c':
+                    RMQ_CLUSTER = self.args[i+1]
+                    
+            except (IndexError, ValueError):
+                self.usage()
+            i+=1
+
+        # sspl_setup_consul script install consul in dev env and checks if consul process is running 
+        # on prod. For node replacement scenario consul will not be running on the new node. But,
+        # there will be two instance of consul running on healthy node. When new node is configured
+        # consul will be brought back on it. We are using VIP to connect to consul. So, if consul
+        # is not running on new node, we dont need to error out. So, need to skip this step for 
+        # node replacement case
+        # Onward LDR_R2, consul will be abstracted out and it won't exit as hard dependeny of SSPL
+        if PRODUCT_NAME == "LDR_R1":
+            # setup consul if not running already
+            if not os.path.exists(REPLACEMENT_NODE_ENV_VAR_FILE):
+                SSPLSetup._call_script(f"{self._script_dir}/sspl_setup_consul", ["-e", f"{ENVIRONMENT}"])
+
+        # Install packages which are not available in YUM repo, from PIP
+        pip_cmd = f"python3 -m pip install -r {SSPL_BASE_DIR}/low-level/requirements.txt"
+        subprocess.call(pip_cmd.split(" "), shell=False)
```

Comment:
collect return code and print appropriate error number on failure and exit.

Reply:
subprocess.call will show execution steps as well in the terminal. if any error comes we can see it directly on the terminal.
Changed the code to get the return code.
Done